django_app/redbox_app/redbox_core/views.py

```diff
@@ -180,12 +179,16 @@ def post_message(request: HttpRequest) -> HttpResponse:
         for message in ChatMessage.objects.all().filter(chat_history=session)
     ]
     core_api = CoreApiClient(host=settings.CORE_API_HOST, port=settings.CORE_API_PORT)
-    output_text = core_api.rag_chat(message_history, request.user.get_bearer_token())
+    response_data = core_api.rag_chat(message_history, request.user.get_bearer_token())
 
-    # save LLM response
-    llm_message = ChatMessage(chat_history=session, text=output_text, role=ChatRoleEnum.ai)
+    llm_message = ChatMessage(chat_history=session, text=response_data.output_text, role=ChatRoleEnum.ai)
     llm_message.save()
 
+    files: list[File] = [
```

Comment:
```python
files: list[File] = File.objects.filter(core_file_uuid__in=[doc.file_uuid or doc in response_data.source_documents], user=request.user)
```

this will be a fair bit faster

Reply:
Fixed.